src/main/java/pw/itr0/kaba/encrypt/PasswordBasedEncrypter.java

```diff
@@ -12,27 +17,61 @@
  * @since 1.0.0
  */
 public interface PasswordBasedEncrypter {
+
     /**
-     * Encrypt {@code secret} based on password.
+     * Returns {@link Cipher} to encrypt bytes with {@code password}.
      *
      * @param password encrypting password
+     * @return {@link Cipher} to be used on encrypt
+     * @throws InvalidKeyException if the {@code password} does not satisfy password specification of using encryption algorithm,
+     *                             or encryption algorithm using specified password length is not usable on current JDK
+     */
+    Cipher getEncryptionCipher(char[] password) throws InvalidKeyException;
+
+    /**
+     * Encrypt {@code secret} based on password.
+     *
      * @param secret   secret bytes
+     * @param password encrypting password
      * @return encrypted secret
-     * @throws GeneralSecurityException on encryption failure
+     * @throws InvalidKeyException       if the {@code password} does not satisfy password specification of using encryption algorithm,
+     *                                   or encryption algorithm using specified password length is not usable on current JDK
+     * @throws BadPaddingException       if the {@code secret} is not properly padded
+     * @throws IllegalBlockSizeException if the {@code secret} has wrong length
      */
-    byte[] encrypt(char[] password, byte[] secret) throws GeneralSecurityException;
+    default byte[] encrypt(byte[] secret, char[] password)
+            throws InvalidKeyException, BadPaddingException, IllegalBlockSizeException {
+        return getEncryptionCipher(password).doFinal(secret);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method call to 'getEncryptionCipher' needs "this.".](https://www.codacy.com/app/ryotan/kaba-vault/file/2686265517/issues/source?bid=3298589&fileBranchId=3332567#l44)